src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java

```diff
@@ -220,6 +226,73 @@ public void update(int index,
               });
    }
    
+   private void createResizeColumn()
+   {
+      SafeHtmlRenderer<String> resizerRenderer =
+         new AbstractSafeHtmlRenderer<String>()
+         {
+            @Override
+            public SafeHtml render(String object)
+            {
+               SafeHtmlBuilder sb = new SafeHtmlBuilder();
+               sb.appendHtmlConstant(object);
+               return sb.toSafeHtml();
+            }
+         };
+      objectResizeColumn_ = new Column<RObjectEntry, String>(
+         new ClickableTextCell(resizerRenderer))
+         {
+            @Override
+            public String getValue(RObjectEntry object)
+            {
+               // We don't have ready access to the DOM (still under
+               // construction at this point), so we use a window method here.
+               return "<div class=\"" + style_.colResizer() + "\" " +
+                      "onmousedown=\"rstudio_beginResize(this, " +
+                      "\'" + style_.nameCol() + "'" +
+                      ");\"></div>";
+            }
+         };
+   }
+   
+   private final native void exportResizer() /*-{
+      $wnd.rstudio_beginResize = function(ele, clazz) {
+
+         // Locate the element to resize
```

Comment:
Should we double-check that the user has clicked the primary mouse button (e.g. avoid resizing with middle-click or right-click)?

Reply:
Yes, we should -- all taken care of now.